[Batch Mode] <rdar://40225790> Output a remark from the driver when in batch mode

davidungar opened the pull request:
<!-- What's in this pull request? -->
Facilitate checking for batch mode compilation in the build log by having the driver output a remark when in batch mode.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->


<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

Commit:
Batch mode driver remark

davidungar commented:
@swift-ci please clean smoke test

jrose-apple commented on test/Driver/batch_mode_remark.swift:
```diff
@@ -0,0 +1,12 @@
+//
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift  2>&1 | %FileCheck %s
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift  2>&1 | %FileCheck %s
+// RUN: (%swiftc_driver -enable-batch-mode        -disable-batch-mode  %S/../Inputs/empty.swift  2>&1; echo stop-empty-complaint) | %FileCheck %s
```
I think there's a FileCheck option for this…looks like `-allow-empty`.

jrose-apple commented on test/Driver/batch_mode_remark.swift:
```diff
@@ -0,0 +1,12 @@
+// Ensure that driver issues a remark iff in batch mode.
+//
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift  2>&1 | %FileCheck %s
```
You should either use `%target-build-swift` for these invocations or make sure that the job doesn't actually get executed (using `-###` or something), so that the test works correctly when only building a cross-compiler. (Which is something we do for iOS sometimes within Apple, but I don't remember if any of the public bots do it.)

davidungar replied:
Added the -###.

jrose-apple commented on test/Driver/batch_mode_remark.swift:
```diff
@@ -0,0 +1,12 @@
+// Ensure that driver issues a remark iff in batch mode.
+//
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift  2>&1 | %FileCheck %s
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift  2>&1 | %FileCheck %s
```
What's the difference between this line and the previous one?

Also, would it make sense to test what no arguments at all does here?

davidungar replied:
D'Oh! That's what I meant to do there. Fixed.

jrose-apple reviewed (commented):
Glad it seems to be this simple! Did you get a chance to test it in Xcode? (I'm sorry I didn't, I ended up feeling sickish on Friday.)

Commit:
test fixes

davidungar commented:
@swift-ci -please smoke test

davidungar commented:
@jrose-apple Sorry you felt sick. Hope you're better. Have not tested in Xcode yet, but will do so tonight.
Meanwhile, have addressed your review (tx!) and pushed. Smoke tests seem to be failing for an irrelevant reason.

davidungar commented:
@jrose-apple Tested in Xcode, both build systems, no problem.

davidungar commented:
@swift-ci -please smoke test

davidungar commented:
@swift-ci -please smoke test

davidungar commented:
@swift-ci please smoke test

jrose-apple commented on test/Driver/batch_mode_remark.swift:
```diff
@@ -0,0 +1,12 @@
+//
+// RUN: %swiftc_driver -whole-module-optimization -enable-batch-mode   %S/../Inputs/empty.swift -### 2>&1 | %FileCheck %s
+// RUN: %swiftc_driver                                                 %S/../Inputs/empty.swift -### 2>&1 | %FileCheck %s
+// RUN: %swiftc_driver -enable-batch-mode        -disable-batch-mode   %S/../Inputs/empty.swift -### 2>&1 | %FileCheck -allow-empty %s
```
I'm a little surprised only one of these is empty. Are the others emitting some *other* diagnostic?

jrose-apple reviewed (approved):
Great, glad it was this easy!

Commit:
Cleaned up test

Commit:
Update tests

davidungar commented:
@swift-ci please smoke test

davidungar commented:
@swift-ci please smoke test OS X platform
